data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -84,26 +91,39 @@ ReceiveMessageRequest createReceiveMessageRequest() {
     }
 
     private Map<Message, S3EventNotification.S3EventNotificationRecord> getS3MessageEventNotificationRecordMap(final List<Message> sqsMessages) {
-        return sqsMessages.stream().collect(Collectors.toMap(message -> message, this::convertS3EventMessages));
+        return sqsMessages.stream()
+                .collect(HashMap::new,
+                        (messageRecordMap, message) -> messageRecordMap.put(message, convertS3EventMessages(message)),
+                        HashMap::putAll);
     }
 
     S3EventNotification.S3EventNotificationRecord convertS3EventMessages(final Message message) {
-        return S3EventNotification.parseJson(message.body()).getRecords().get(0);
+        try {
+            return S3EventNotification.parseJson(message.body()).getRecords().get(0);
+        } catch (SdkClientException e) {
+            LOG.error("Invalid JSON string in message body of {}", message.messageId());
```

Comment:
I think it would be good to include either the exception or the exception message in the log. Note that to log the exception, you just add `, e);` to the end of the arguments list. If you want to include only the exception message, you will need to add `{}` to the format string.